File: user-srv-v2/internal/dto/suitabilitytest.go
package dto

type (
	SuitabilityTestRequest struct {
		Grade       string `json:"grade"`
		Score       int    `json:"score"`
		Version     string `json:"version"`
		ReviewDate  string `json:"reviewDate"`
		ExpiredDate string `json:"expiredDate"`
	}

	SuitabilityTestResponse struct {
		Grade       string `json:"grade"`
		Score       int    `json:"score"`
		Version     string `json:"version"`
		ReviewDate  string `json:"reviewDate"`
		ExpiredDate string `json:"expiredDate"`
	}
)

File: user-srv/src/Pi.User/Pi.User.Domain/AggregatesModel/SuitabilityTestAggregate/SuitabilityTest.cs
using Pi.Common.SeedWork;

namespace Pi.User.Domain.AggregatesModel.SuitabilityTestAggregate;

public class SuitabilityTest(Guid id) : Entity<Guid>(id), IAggregateRoot
{
    public required string Grade { get; set; }
    public int Score { get; set; }
    public required string Version { get; set; }
    public DateTime ReviewDate { get; set; }
    public DateTime ExpiredDate { get; set; }
    public Guid UserId { get; init; }
}